snuba/migrations/snuba_migrations/metrics/templates.py

```diff
@@ -174,6 +179,31 @@ def get_forward_bucket_table_dist(
     retention_days
 """
 
+MATVIEW_STATEMENT_POLYMORPHIC_TABLE_V2 = """
+SELECT
+    org_id,
+    project_id,
+    metric_id,
+    arrayJoin([10,60,3600,86400]) as granularity,
+    tags.key,
+    tags.value,
+    toDateTime(granularity * intDiv(toUnixTimestamp(timestamp), granularity)) as timestamp,
+    retention_days,
+    %(aggregation_states)s
+FROM %(raw_table_name)s
+WHERE materialization_version = 4
```

Comment:
Oh wait, the only difference here is the materialization_version? 
Sorry I misunderstood the problem this morning before reading this. 
If the matview template is constant but the only thing we change is the materialization version I think it would make sense to use the template. This is the same change we will make over and over again. 
Basically we already do that for the `%(raw_table_name)s` one line above.

I thought we were making actual functional changes to the matview that would not be backwards compatible. 

I suggest we turn this line into `materialization_version = %(materialization_version)s` on the template. We can create a new template to avoid changing the existing migrations (in hindsight we should have made materialization_version a parameter to begin with).